policies.go

```diff
@@ -395,6 +395,18 @@ func ShuffleReplicas() func(*tokenAwareHostPolicy) {
 	}
 }
 
+// NonLocalReplicasFallback enables fallback to replicas that are not considered local.
+//
+// TokenAwareHostPolicy used with DCAwareHostPolicy fallback first selects replicas by partition key in local DC, then
+// falls back to other nodes in the local DC. Enabling NonLocalReplicasFallback causes TokenAwareHostPolicy
+// to first select replicas by partition key in local DC, then replicas by partition key in remote DCs and fall back
+// to other nodes in local DC.
+func NonLocalReplicasFallback() func(policy *tokenAwareHostPolicy) {
```

Comment:
This one is exported but it returns a function with an unexported argument.
How is this intended to be used?

Reply:
It is a [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) pattern that is used by `TokenAwareHostPolicy` already for `ShuffleReplicas()` option, so I copied the style for another option in that function to keep the code consistent. It could be used as an argument to `TokenAwareHostPolicy` like this:

```go
configuration.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(
		gocql.DCAwareRoundRobinPolicy(datacenter),
		gocql.ShuffleReplicas(),
		gocql.NonLocalReplicasFallback(),
	)
```

I'd prefer `tokenAwareHostPolicy` type to be exported as well as rename `TokenAwareHostPolicy()` to `NewTokenAwareHostPolicy()`, but that would break backward compatibility unfortunately.